tests/helper/labels.go

```diff
@@ -5,17 +5,29 @@ import (
 )
 
 const (
+	// LabelCluster is the default label, indicating tests that want to run with a cluster.
+	// If no label is set, tests are assumed to be running in cluster mode.
+	LabelCluster   = "cluster"
 	LabelNoCluster = "nocluster"
 	LabelUnauth    = "unauth"
 	LabelPodman    = "podman"
 )
 
 func NeedsCluster(labels []string) bool {
 	for _, label := range labels {
+		if label == LabelCluster {
+			return true
+		}
 		if label == LabelNoCluster {
 			return false
 		}
 		if label == LabelPodman {
+			// Check if there is any "cluster" label
+			for _, l := range labels {
+				if l == LabelCluster {
+					return true
```

Comment:
As I understand, a test with Podman and Cluster labels will fail on a Podman only test environment.

For example:
```
make test-integration-podman 

[...]

[FAILED] [3.165 seconds]
odo describe component command tests [BeforeEach]
/home/phmartin/Documents/gits/odo/tests/integration/cmd_describe_component_test.go:21
  when creating a component
  /home/phmartin/Documents/gits/odo/tests/integration/cmd_describe_component_test.go:106
    when running odo dev (podman=true) [podman]
    /home/phmartin/Documents/gits/odo/tests/integration/cmd_describe_component_test.go:228
      should describe the component in dev mode (experimental=false) [cluster]
      /home/phmartin/Documents/gits/odo/tests/integration/cmd_describe_component_test.go:254

  Begin Captured GinkgoWriter Output >>
    Created dir: /tmp/543739364
    Created dir: /tmp/2825039718
    Setting KUBECONFIG=/tmp/2825039718/config
    Creating a new project: cmd-describe-component-test254cre
    Running oc with args [oc new-project cmd-describe-component-test254cre] and odo env: []
    [oc] Unable to connect to the server: dial tcp 192.168.1.194:6443: connect: no route to host
    Setting current dir to: 
  << End Captured GinkgoWriter Output

  No future change is possible.  Bailing out early - after 3.164s.
  Running oc with args [oc new-project cmd-describe-component-test254cre] and odo env: []
  Expected
      <int>: 1
  to match exit code:
      <int>: 0
  In [BeforeEach] at: /home/phmartin/Documents/gits/odo/tests/helper/helper_cmd_wrapper.go:101

  There were additional failures detected after the initial failure.  Here's a summary - for full details run Ginkgo in verbose mode:
    [FAILED] in [AfterEach] at /home/phmartin/Documents/gits/odo/tests/helper/helper_filesystem.go:94
```

Reply:
For the record, as discussed together, we'll need to think about how to run tests with both Podman and Cluster modes consistently.
I remember what I initially wanted to test: the behavior of odo describe component --name with the experimental mode disabled (default cluster mode), but with odo dev only running on Podman => odo describe component --name should check the cluster and report that it could not find any component in the namespace.
For now, I have updated this PR accordingly, just checking that an error is returned.
Done in [`2279410` (#6431)](https://github.com/redhat-developer/odo/pull/6431/commits/22794101590ab3824090ab11f06f710f6b967e33#diff-505f09abcdb40224f164257b1ebb2062685fa6c461761f9c3fbce9d97b23ace1).